### PR TITLE
Adds plugin writer docs on Master/Detail pattern

### LINKED ISSUE #2981: As a plugin writer, I have docs on how the Master/Detail pattern works

### PR DESCRIPTION
**Pull request by bmbouter**
closes #2981

### CODE REVIEW

File: docs/plugins/plugin-writer/concepts/index.rst [hunk 1]
```diff
@@ -39,6 +39,69 @@ the box.
...
+    >>> type(my_file_remote)  # We queried the detail type so we expect that type of instance
+    pulp_file.app.models.FileRemote
+
+    >>> my_file_remote.policy = "streamed"  # The detail object acts like it has all the attrs
+    >>> my_file_remote.save()  # Django's multi-table inheritance handles where to put things
+
+    >>> my_master_remote = my_file_remote.master  # the `master` attr gives you the master instance
+
+    >>> type(my_master_remote)  # Let's confirm this is the Master model type
+    pulpcore.app.models.repository.Remote
```
Comment (mdellweg) [inline docs/plugins/plugin-writer/concepts/index.rst:83]: Did you want to show that `policy` is a master-model attribute?

**Review by mdellweg — COMMENTED** [review]
Do you want to mention, that it is possible to subclass a detail model and that `cast` will always derive the most recent descendent?

**Comment by bmbouter** [comment]
@mdellweg I pushed revisions along with some new content, can you re-review? @ipanova @gerrod3 @dralley can either of you be second reviewer?

File: docs/plugins/plugin-writer/concepts/index.rst [hunk 2]
```diff
@@ -39,6 +39,90 @@ the box.
...
+inherits from ``MasterModel`` which identifies it as a Master model, and defines many fields. Then
+pulp_file defines the `FileRemote <https://github.com/pulp/pulp_file/blob/
+68cdbde4f7f9609d987ec4e6694810e6085288db/pulp_file/app/models.py#L46>`_ which is a Detail model. The
+Detail model defines a ``TYPE`` class attribute and is a subclass of a Master model.
+
+Typically Master models are provided by pulpcore, and Detail models by plugins, but this is not
+strictly required. Here is a list of the Master models pulpcore provides:
+
+* ``pulpcore.plugin.models.AlternateContentSource``
+* ``pulpcore.plugin.models.BaseDistribution``
```
Comment (mdellweg) [inline docs/plugins/plugin-writer/concepts/index.rst:72]: This is deprecated we should not mention it.

Comment (bmbouter) [reply docs/plugins/plugin-writer/concepts/index.rst:72]: I was also wondering what this was doing here. Were we supposed to remove this for 3.20 but forgot? If so we should make a 3.20 milestone and put a ticket to remove it on there. Lmk if that's the case and I can do the bookkeeping if that's helpful.

**Comment by bmbouter** [comment]
@mdellweg thanks! new version pushed.

File: docs/plugins/plugin-writer/concepts/index.rst (hunk 2 above)
Comment (mdellweg) [reply docs/plugins/plugin-writer/concepts/index.rst]: When trying to remove it there were issues with migrations i keep forgetting about.

**Review by dralley — APPROVED** [review]

**Pull request merged**